### PR TITLE
fix hadoop Path for names with colon

### PR DESCRIPTION
**Pull request by singhravidutt**

### CODE REVIEW

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java [hunk 1]
```diff
@@ -54,6 +56,20 @@ public void before() throws IOException {
     postCreateInit();
   }
 
+  @Test
+  public void verifyHadoopPath() throws Exception {
+    GoogleHadoopFileSystem eagerFs =
+        GoogleHadoopFileSystemTestHelper.createInMemoryGoogleHadoopFileSystem();
+    String fileNameWithColon = "empty:file";
+    FSDataOutputStream stream =
```
Comment (medb) [inline gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java:64]: nit: use try-with-resource to close stream automatically

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java [hunk 2]
```diff
@@ -54,6 +56,20 @@ public void before() throws IOException {
...
+        eagerFs.create(new Path(/* schema= */ null, /* authority= */ null, fileNameWithColon));
+    stream.close();
+    FileStatus[] fileStatus =
+        eagerFs.listStatus(new Path(GoogleHadoopFileSystemTestHelper.IN_MEMORY_TEST_BUCKET));
+    assertThat(fileStatus.length).isEqualTo(1);
```
Comment (medb) [inline gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java:69]: nit: is there `hasLength` assert method?

**Review by medb — APPROVED** [review]

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java (hunk 1 above)
Comment (singhravidutt) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java]: Wouldn't even need it closing it immediately as it's an empty file.

File: gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java (hunk 2 above)
Comment (singhravidutt) [reply gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java]: updated.

**Review by vinayakumarb — APPROVED** [review]

**Pull request merged**